From ba909f59ac5da3c69a2b166a8a9d5c1b14d3a6ad Mon Sep 17 00:00:00 2001
From: Phillip Lougher <phillip@squashfs.org.uk>
Date: Mon, 31 Jul 2017 20:37:54 +0100
Subject: [PATCH 117/120] mksquashfs: add pseudo definition format to the help
 text

Display the format in the help text, and not only on error when
the format is not recognised by the parser.

Signed-off-by: Phillip Lougher <phillip@squashfs.org.uk>
---
 squashfs-tools/mksquashfs.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/squashfs-tools/mksquashfs.c b/squashfs-tools/mksquashfs.c
index dae0f01..75b3577 100644
--- a/squashfs-tools/mksquashfs.c
+++ b/squashfs-tools/mksquashfs.c
@@ -3,7 +3,7 @@
  * filesystem.
  *
  * Copyright (c) 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011,
- * 2012, 2013, 2014
+ * 2012, 2013, 2014, 2017
  * Phillip Lougher <phillip@squashfs.org.uk>
  *
  * This program is free software; you can redistribute it and/or
@@ -5619,6 +5619,14 @@ printOptions:
 				"definition\n");
 			ERROR("-pf <pseudo-file>\tAdd list of pseudo file "
 				"definitions\n");
+			ERROR("\t\t\tPseudo definitions should be of the "
+				"format\n");
+			ERROR("\t\t\t\tfilename d mode uid gid\n");
+			ERROR("\t\t\t\tfilename m mode uid gid\n");
+			ERROR("\t\t\t\tfilename b mode uid gid major minor\n");
+			ERROR("\t\t\t\tfilename c mode uid gid major minor\n");
+			ERROR("\t\t\t\tfilename f mode uid gid command\n");
+			ERROR("\t\t\t\tfilename s mode uid gid symlink\n");
 			ERROR("-sort <sort_file>\tsort files according to "
 				"priorities in <sort_file>.  One\n");
 			ERROR("\t\t\tfile or dir with priority per line.  "
-- 
2.8.3

